js/main.js

```diff
@@ -175,46 +180,81 @@ function newDeck(event) {
     cards: [],
     deckView: 'deck-' + data.numberOfDecks
   });
-  appendDeck(data.deck[0]);
+  if (!document.querySelector('#deck-row').children) {
+    appendDeck(data.deck[0]);
+  }
   if (document.querySelector('#no-decks-available')) {
 
     document.querySelector('#no-decks-available').remove();
   }
 }
 document.addEventListener('click', newDeck);
 
-function addCard(event) {
-  if ($cardRow.children) {
-    var srcImage;
-    var $modalImage = document.querySelector('#add-image>img');
-    var cards = $cardRow.getElementsByTagName('img');
-    var $add = document.querySelector('#add-image');
-    for (var i = 0; i < cards.length; i++) {
-      if (event.target === cards[i]) {
+function addDeleteCard(event) {
+  var $modalImage = document.querySelector('#add-image>img');
+  if (document.querySelector('#search-id').className === 'tab-view') {
```

Comment:
Is this to determine what the current view is? If so, you should be updating what the current view is in localStorage, like you did in `code-journal`. That way you don't need to query the DOM to determine what to do here.